### ISSUE TITLE
Migrate `configuration` command into datalad proper

### ISSUE THREAD

**Issue by mih** [datalad/datalad#5489]
It is presently provided by an extension and has to be installed separately. However, I find it rather handy, and consider it a candidate for inclusion into core.

https://github.com/mih/datalad-mihextras

https://datalad-mihextras.readthedocs.io/en/latest/generated/datalad.api.x_configuration.html

It was briefly already available, but was removed https://github.com/datalad/datalad/pull/5370

**Comment by mih** [comment]
Opinions have not changed. Taking it off the 0.15 milestone.

**Comment by yarikoptic** [comment]
> The alternatively requested renaming of parameters and their values would only establish superficial alignment of the APIs

Re `2` and `3` which are IMHO a singular aspect. As we have discussed on the call, renaming could as well be done at `ConfigManager` level  (`where` -> `scope`; `dataset` -> `branch`) through a deprecation cycle (e.g. add for 0.15.0 and deprecate/remove old style in 0.16.0 or later ;) ), thus gently transitioning/preparing users to use `configuration` options and not requiring them to remember what `option=value` to use with what API endpoint. IMHO it would not be a "superficial alignment" since AFAIK they refer to exactly the same thing, and I do like `scope` and `branch` better (even surprised I did not complain/suggest it whenever `where="dataset"` was introduced in #922).  And FWIW I don't care about `4` (`system`-level) and also do not see how to avoid `1` in a reasonable effort.
